hazelcast-jet-core/src/main/java/com/hazelcast/jet/Sinks.java

```diff
@@ -58,6 +64,204 @@ private Sinks() {
+     *
+     *  @param mapName name of the map
+     *  @param toKeyFn  function that extracts key from the input item
+     *  @param updateFn function contains the update logic.The function
+     *                 will be called with {@code updateFn.apply(E item, V oldValue)}
+     *  @param <E> type of input item
+     *  @param <K> type of the key
+     *  @param <V> type of the value
+     */
+    public static <E, K, V> Sink<E> updatingMap(
+            @Nonnull String mapName,
+            @Nonnull DistributedFunction<E, K> toKeyFn,
+            @Nonnull DistributedBiFunction<E, V, V> updateFn
+    ) {
+        return fromProcessor("updatingMapSink(" + mapName + ')', updateMapP(mapName, toKeyFn, updateFn));
+    }
+
+    /**
+     * Returns a sink that updates the items it receives into a Hazelcast
+     * {@code IMap} with the specified name and update function in a
+     * remote cluster identified by the supplied {@code ClientConfig}.
+     *
+     * <p>
+     * The {@code updateFn}  will be called with the input item as {@code item}
+     * and the {@code oldValue}, which is the associated value of the extracted key
+     * in Hazelcast Map. The return  value of the {@code updateFn} will be
+     * the new value of the key.
+     * </p>
+     *
+     *  @param mapName name of the map
+     *  @param clientConfig client configuration to be able to connect remote
+     *                      cluster
+     *  @param toKeyFn  function that extracts key from the input item
+     *  @param updateFn function contains the update logic. The function
+     *                 will be called with {@code updateFn.apply(E item, V oldValue)}
+     *  @param <E> type of input item
+     *  @param <K> type of the key
+     *  @param <V> type of the value
+     */
+    public static <E, K, V> Sink<E> remoteUpdatingMap(
+            @Nonnull String mapName,
+            @Nonnull ClientConfig clientConfig,
+            @Nonnull DistributedFunction<E, K> toKeyFn,
+            @Nonnull DistributedBiFunction<E, V, V> updateFn
+
+    ) {
+        return fromProcessor("remoteUpdatingMapSink(" + mapName + ')',
+                updateRemoteMapP(mapName, clientConfig, toKeyFn, updateFn));
+    }
+
+    /**
+     * Returns a sink that updates the items it receives into a Hazelcast
+     * {@code IMap} with the specified name using {@code EntryProcessor}.
+     *
+     * <p>
+     * This method internally uses {@code IMap.submitToKey() } method to
+     * invoke provided {@code EntryProcessor} on the key which does not
+     * utilize batching. There will be an invocation for each key.
+     * Main purpose of this method is to support {@code Offloadable}
+     * entry processors. If you don't have an {@code Offloadable}
+     * entry processor please use one of the following to implement
+     * your update/merge logic. The following methods will utilize
+     * batching while interacting with Hazelcast Cluster which is
+     * more efficient.
```

Comment:
I rewrote this as follows:

```
Returns a sink that updates the items it receives into a Hazelcast
{@code IMap} with the specified name using an {@code EntryProcessor}
per item.
<p>
Internally, the sink uses an entry processor per key is submitted. This method is not
as efficient as {@see Sinks#updatingMap} or {@see Sinks#mergingMap}
because it submits one {@link EntryProcessor} per key rather than in batches.
<p>
The supplied EntryProcessor can be made {@link Offloadable} to avoid
blocking the partition thread during large update operations.
```